src/main/scala/vsys/state/opcdiffs/AssertOpcDiff.scala

```diff
@@ -0,0 +1,103 @@
+package vsys.state.opcdiffs
+
+import com.google.common.primitives.Longs
+import scorex.account.Address
+import scorex.transaction.ValidationError
+import scorex.transaction.ValidationError.{ContractDataTypeMissMatch, ContractInvalidCaller, ContractInvalidOPCData, ContractInvalidSigner, GenericError}
+import vsys.contract.{DataEntry, DataType, ExecutionContext}
+
+import scala.util.{Left, Right}
+
+object AssertOpcDiff {
+
+  def gtEq0(v: DataEntry): Either[ValidationError, OpcDiff] = {
+    if (v.dataType == DataType.Amount && Longs.fromByteArray(v.data) >= 0)
+      Right(OpcDiff.empty)
+    else
+      Left(GenericError(s"Invalid Assert (gteq0): Value ${Longs.fromByteArray(v.data)} is negative"))
+  }
+
+  def ltEq(v1: DataEntry, v2: DataEntry): Either[ValidationError, OpcDiff] = {
+    if (v1.dataType == DataType.Amount && v2.dataType == DataType.Amount
+      && Longs.fromByteArray(v1.data) <= Longs.fromByteArray(v2.data))
+      Right(OpcDiff.empty)
+    else
+      Left(GenericError(s"Invalid Assert (lteq0): Value ${Longs.fromByteArray(v2.data)} is larger than $v1"))
+  }
+
+  def ltInt64(m: DataEntry): Either[ValidationError, OpcDiff] = {
+    if (m.dataType == DataType.Amount && Longs.fromByteArray(m.data) <= Long.MaxValue)
```

Comment:
isn't "Longs.fromByteArray(m.data) <= Long.MaxValue" always true?

Reply:
yes. should double check the redundance code